recipes/cad1/task1/baseline/enhance.py

```diff
@@ -0,0 +1,304 @@
+            When `device` is different from `mix.device`, only local computations will
+            be on `device`, while the entire tracks will be stored on `mix.device`.
+
+    Returns:
+        torch.Tensor: estimated sources
+
+    Based on https://pytorch.org/audio/main/tutorials/hybrid_demucs_tutorial.html
+    """
+    device = mix.device if device is None else torch.device(device)
+    mix = torch.as_tensor(mix, device=device)
+
+    if mix.ndim == 1:
+        # one track and mono audio
+        mix = mix.unsqueeze(0)
+    elif mix.ndim == 2:
+        # one track and stereo audio
+        mix = mix.unsqueeze(0)
+
+    batch, channels, length = mix.shape
+
+    chunk_len = int(sample_rate * segment * (1 + overlap))
+    start = 0
+    end = chunk_len
+    overlap_frames = overlap * sample_rate
+    fade = Fade(fade_in_len=0, fade_out_len=int(overlap_frames), fade_shape="linear")
+
+    final = torch.zeros(batch, len(model.sources), channels, length, device=device)
+
+    while start < length - overlap_frames:
+        chunk = mix[:, :, start:end]
+        with torch.no_grad():
+            out = model.forward(chunk)
+        out = fade(out)
+        final[:, :, :, start:end] += out
+        if start == 0:
+            fade.fade_in_len = int(overlap_frames)
+            start += int(chunk_len - overlap_frames)
+        else:
+            start += chunk_len
+        end += chunk_len
+        if end >= length:
+            fade.fade_out_len = 0
+
+    return final.cpu().detach().numpy()
+
+
+def get_device(device: str) -> tuple:
+    """Get the Torch device.
+
+    Args:
+        device (str): device type, e.g. "cpu", "gpu0", "gpu1", etc.
+
+    Returns:
+        torch.device: torch.device() appropiate to the hardware available.
+        str: device type selected, e.g. "cpu", "cuda".
+    """
+    if device is None:
+        if torch.cuda.is_available():
+            return torch.device("cuda"), "cuda"
+        return torch.device("cpu"), "cpu"
+
+    elif device.startswith("gpu"):
+        device_index = int(device.replace("gpu", ""))
+        if device_index > torch.cuda.device_count():
+            raise ValueError(f"GPU device index {device_index} is not available.")
+        return torch.device(f"cuda:{device_index}"), "cuda"
+
+    elif device == "cpu":
+        return torch.device("cpu"), "cpu"
+
+    else:
+        raise ValueError(f"Unsupported device type: {device}")
+
+
+def map_to_dict(sources: np.ndarray, sources_list: List[str]) -> Dict:
+    """Map sources to a dictionary."""
+    audios = dict(zip(sources_list, sources))
+
+    signal_stems = {}
+    for source in sources_list:
+        audio = audios[source]
+        signal_stems[f"left_{source}"] = audio[0]
+        signal_stems[f"right_{source}"] = audio[1]
+
+    return signal_stems
+
+
+def decompose_signal(
+    model: torch.nn.Module, signal: np.ndarray, sample_rate: int, device: torch.device
+) -> Dict[str, np.ndarray]:
+    """Decompose signal into 8 stems."""
```

Comment:
Could do with docstrings here.
```suggestion
    """Decompose signal into 8 stems.
    
    Args:
        model (torch.nn.Module): Torch model.
        signal (np.ndarray): Signal to be decomposed.
        sample_rate (int): Sample frequency.
        device (torch.device): Torch device to use for processing.
        
     Returns:
         Dictionary: Indexed by sources with the associated model as values.
     """
```

If the above is ok, you can apply the "suggestion" here on GitHub, although you may want to then `git pull` locally so that the branch is pulled down and you have the changes on your local computer.